### PR TITLE
Season stats

### PR DESCRIPTION
**Pull request by dborski**
Adds season statistics methods:

- games_ids_by_season
- games_by_season_id
- team_name_by_team_id
- games_by_team_name
- games_by_head_coach
- winning_percentage_by_head_coach
- winningest_coach
- worst_coach
- shooting_percentage_by_team
- most_accurate_team
- least_accurate_team
- tackles_by_team
- most_tackles
- fewest_tackles

### CODE REVIEW

File: lib/season_stats.rb [hunk 1]
```diff
@@ -8,34 +8,95 @@ def initialize(game_teams, games, teams)
...
+      winning_games = value.find_all { |value| value.result == "WIN" }
+      if winning_games.length != 0
+        value = ((winning_games.length.to_f / value.length.to_f) * 100).round(2)
+        winning_percentage_by_head_coach[key] = value
+      else
+        winning_percentage_by_head_coach[key] = 0
+      end
+    end
+    winning_percentage_by_head_coach
+  end
```
Comment (whitneykidd) [inline lib/season_stats.rb:52]: Nice work on this beast of a method

File: lib/season_stats.rb [hunk 2]
```diff
@@ -22,20 +22,30 @@ def games_by_season_id(season_id)
...
+  def winning_percentage_by_head_coach(season_id)
+    games_by_head_coach = games_by_season_id(season_id).group_by { |game| game.head_coach}
+
+    winning_percentage_by_head_coach = {}
+
+    games_by_head_coach.each do |key, value|
+      winning_games = value.find_all {|value| value.result == "WIN"}
+      if winning_games.length != 0
+        value = ((winning_games.length.to_f / value.length.to_f) * 100).round(2)
+        winning_percentage_by_head_coach[key] = value
```
Comment (kiverso) [inline lib/season_stats.rb:46]: I use similar logic to get win percentages in some team_stats methods, could be good to move to a module once all our code is functional.

File: lib/season_stats.rb [hunk 3]
```diff
@@ -24,24 +24,24 @@ def games_by_season_id(season_id)
...
-    games_by_id = games_by_season_id(season_id).group_by {|game| game.team_id}
+    games_by_id = games_by_season_id(season_id).group_by { |game| game.team_id }
     games_by_id.transform_keys(&team_name_by_team_id.method(:[]))
   end
 
-  def winning_percentage_by_head_coach(season_id)
-    games_by_head_coach = games_by_season_id(season_id).group_by { |game| game.head_coach}
+  def games_by_head_coach(season_id)
+    games_by_season_id(season_id).group_by { |game| game.head_coach }
+  end
```
Comment (kiverso) [inline lib/season_stats.rb]: I like this helper method, it is easier to follow when this is outside of the winning percentage method.

File: lib/season_stats.rb [hunk 4]
```diff
@@ -8,34 +8,95 @@ def initialize(game_teams, games, teams)
...
+    games_by_id = games_by_season_id(season_id).group_by { |game| game.team_id }
+    games_by_id.transform_keys(&team_name_by_team_id.method(:[]))
+  end
+
+  def games_by_head_coach(season_id)
+    games_by_season_id(season_id).group_by { |game| game.head_coach }
+  end
+
+  def winning_percentage_by_head_coach(season_id)
+    winning_percentage_by_head_coach = {}
```
Comment (IamNorma) [inline lib/season_stats.rb:41]: Nice job with this method

**Pull request merged**